### DISCUSSION TITLE
Session logging / transcript

### DISCUSSION THREAD

**Discussion by ECon65** [wezterm/wezterm#3210] [Q&A]
Is it possible to configure logging to save printable output from each session to a file (using some log rotation options)?

**Comment by wez** [comment]
Not built into the terminal, but you can use `wezterm record` and `wezterm replay` to capture and replay sessions.

* Inside any terminal run `wezterm record` to start a recording session.
* When you exit the shell in that session you should see a message like:

```
*** Finished recording to /var/tmp/wezterm-recording-sF6B3u.cast.txt
```

The file is an asciicast (compatible with https://asciinema.org/) and can also be replayed using `wezterm replay`.

**Reply by ECon65** [reply]
Hi @wez,

thank you for your answer. This feature is very helpful. I'm only missing a possibility to capture the output without any metadata something like `--raw` provided by [asciinema](https://asciinema.org/docs/usage#:~:text=%2D%2D-,raw,-%2D%20Save%20raw%20STDOUT)

Would it be possible for you to add such option?

**Reply by wez** [reply]
Please file an enhancement issue for that

**Reply by ECon65** [reply]
I've opened an enhancement [#3446](https://github.com/wez/wezterm/issues/3446)

thank you for your kind support

**Reply by ECon65** [reply]
Hi @wez,

thank you for providing the option as documented in [#3446](https://github.com/wez/wezterm/issues/3446). Still I'm not sure whether it works correctly:

- if I use `wezterm replay --cat filename.cast.txt`  it outputs the recorded session without replaying it which is the expected behavior
- if I use `wezterm replay --cat filename.cast.txt > filename.txt` the resulting file still contains more than a raw stdout output.

In addition to that I've noticed a strange error when using `wezterm record`. Whenever I try to paste longer text or text containing special characters to a recorded session the input is messed up.
For example trying to paste `cat $PROFILE | findstr /i "wezterm.cli"` results in `cat $P1_ROF6;1_LE |124;0;16;1_ finds;1_r /i ";42;0;0;0;1_wezterm9;0;0;1_.cli"`
